ops/charm.py

```diff
@@ -1100,7 +1100,12 @@ class CollectStatusEvent(LifecycleEvent):
     way after the end of every hook, it should observe the
     :attr:`collect_app_status <CharmEvents.collect_app_status>` or
     :attr:`collect_unit_status <CharmEvents.collect_unit_status>` event,
-    respectively.
+    respectively. Note that this event is triggered at the end of *every* hook,
+    so any observer of these events will be run on every hook, regardless of
+    whether the charm observes that hook event. For example, even if you don't
+    observe :attr:`config_changed <CharmEvents.config_changed>`, your collect status
+    observer will run at the end of every :attr:`config_changed <CharmEvents.config_changed>`
+    event.
```

Comment:
I feel like we're maybe now going *too* far. There are 3 "every hook" comments in the doc now :)

Maybe:
```suggestion
    respectively. Note that observing the specific hook event is not required. For example, even
    if you don't observe :attr:`config_changed <CharmEvents.config_changed>`, your collect status
    observer will run at the end of every :attr:`config_changed <CharmEvents.config_changed>`
    event, as well as all others.
```

Reply:
Thanks, that's much better